### PR TITLE
OE-258 Conditionaly dropped tables to run this migration properly in a live environment

### PR DESCRIPTION
**Pull request by Vegemash**

### CODE REVIEW

**Comment by marshl** [comment]
You know, instead of having commit messages like "quotes *sigh*", you could just amend the commit by staging the changes and then running "git commit --amend --no-edit"

File: protected/modules/OphTrConsent/migrations/m170907_181843_anaesthetic_type_multiselect.php [hunk 1]
```diff
@@ -19,6 +21,19 @@ public function up()
...
         $this->dropForeignKey('et_ophtrconsent_procedure_anaesthetic_type_id_fk', 'et_ophtrconsent_procedure');
 
+        try {//Drop if exists by another name
+            $this->execute('ALTER TABLE `et_ophtrconsent_procedure` DROP FOREIGN KEY `et_ophtrconsent_procedure_anaesthetic_type_id_fk`;');
+            $this->execute('ALTER TABLE `et_ophtrconsent_procedure` DROP INDEX `et_ophtrconsent_procedure_anaesthetic_type_id_fk` ;');
+        }catch (Exception $e){}
+
+        $this->execute('
+          UPDATE  openeyes.et_ophtrconsent_procedure 
+          SET anaesthetic_type_id = 7
```
Comment (marshl) [inline protected/modules/OphTrConsent/migrations/m170907_181843_anaesthetic_type_multiselect.php]: You may want to explain why "7" and not any of the other numbers

Comment (apigram) [reply protected/modules/OphTrConsent/migrations/m170907_181843_anaesthetic_type_multiselect.php]: Exactly what I was going to say.

File: protected/modules/OphTrConsent/migrations/m170907_181843_anaesthetic_type_multiselect.php [hunk 2]
```diff
@@ -19,6 +21,19 @@ public function up()
 
         $this->dropForeignKey('et_ophtrconsent_procedure_anaesthetic_type_id_fk', 'et_ophtrconsent_procedure');
 
+        try {//Drop if exists by another name
+            $this->execute('ALTER TABLE `et_ophtrconsent_procedure` DROP FOREIGN KEY `et_ophtrconsent_procedure_anaesthetic_type_id_fk`;');
+            $this->execute('ALTER TABLE `et_ophtrconsent_procedure` DROP INDEX `et_ophtrconsent_procedure_anaesthetic_type_id_fk` ;');
+        }catch (Exception $e){}
```
Comment (marshl) [inline protected/modules/OphTrConsent/migrations/m170907_181843_anaesthetic_type_multiselect.php]: perhaps you should be catching CDbExcetion, or whatever would be thrown if the SQL fails. Catching Exception seems a bit smelly

**Comment by marshl** [comment]
bump

**Comment by apigram** [comment]
Bump

File: protected/modules/OphTrConsent/migrations/m170907_181843_anaesthetic_type_multiselect.php (hunk 1 above)
Comment (Vegemash) [reply protected/modules/OphTrConsent/migrations/m170907_181843_anaesthetic_type_multiselect.php]: It's for no anaesthetic

**Review by apigram — APPROVED** [review]

**Pull request merged**